Dev

SerjePerm opened the pull request:
Sprint #9 PR #1

SerjePerm committed:
sprint #9 commit #1

SerjePerm committed:
sprint #9 commit #2

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,11 +1,63 @@
+    companion object {
+        const val searchField = "SEARCH_FIELD"
+        const val searchFieldDefault = ""
+    }
```
`companion object` принято объявлять внизу класса согласно Kotlin Coding Convention (https://kotlinlang.org/docs/coding-conventions.html#class-layout)

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,11 +1,63 @@
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) { }
+            override fun afterTextChanged(s: Editable?) { }
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                if (s.isNullOrEmpty()) ivClear.visibility = View.GONE else ivClear.visibility = View.VISIBLE
```
Можно использовать экстеншн проперти `isVisible`:
`ivClear.isVisible = s.isNullOrEmpty()`

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -1,18 +1,61 @@
+        val frameUserLicense = findViewById<FrameLayout>(R.id.frameUserLicense)
+        val frameDarkTheme = findViewById<FrameLayout>(R.id.frameDarkTheme)
+        val switchDark = findViewById<Switch>(R.id.switchDark)
+        if (isDarkTheme()) switchDark.isChecked = true
```
Лучше написать `switchDark.isChecked = isDarkTheme()`, чтобы в любом случае значение на старте активити переопределялось

ArturNurtdinov commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -1,19 +1,54 @@
+
+    <Space
+        android:layout_width="match_parent"
+        android:layout_height="20dp"/>
```
Чтобы не нагружать парсинг лишней вью `Space` это можно сделать с помощью `marginTop` для вью которая ниже

ArturNurtdinov commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -1,19 +1,54 @@
+            android:inputType="text"
+            android:lines="1"
+            android:textColor="@color/lightBlack"
+            android:background="@android:color/transparent"
```
Эти свойства тоже можно вынести в стиль

SerjePerm committed:
sprint #9 commit #3
исправление ошибок/замечаний:

1. кнопка "Очистить" поверх поля ввода в одном FrameLayout
2. companion object перенесён вниз класса
3. установка видимости кнопки "Очистить" реализовано через свойство-расширение ivClear.isVisible = s.isNullOrEmpty().not()
4. исправлено переопределение тёмной темы на switchDark.isChecked = isDarkTheme()
5. в вёрстке ActivitySearch убран Space, заменён на marginTop в LinearLayout
6. свойства поля поиска etSearch вынесены в стиль @style/searchEditText

ArturNurtdinov commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -10,45 +10,40 @@
-            android:lines="1"
-            android:textColor="@color/lightBlack"
-            android:background="@android:color/transparent"
             style="@style/searchEditText"/>
```
Для EditText еще понадобится paddingEnd, чтобы при вводе длинной строки крестик не накладывался на текст

SerjePerm replied:
согласен, исправлю, так сначала и сделал, но меня смутила задача: кнопка "Очистить" поверх поля ввода - получается, что как бы требуется наложение одного элемента на другого

ArturNurtdinov replied:
В FrameLayout как раз это наложение и реализуется. Если использовать Linear или Relative то наложения не будет
Но если есть наложение, надо учитывать, что текст поля ввода может залезть под кнопку очистить и будет некрасиво
